src/bin/server.rs

```diff
@@ -29,9 +29,9 @@ pub async fn main() -> mini_redis::Result<()> {
     Ok(())
 }
 
-#[derive(StructOpt, Debug)]
-#[structopt(name = "mini-redis-server", version = env!("CARGO_PKG_VERSION"), author = env!("CARGO_PKG_AUTHORS"), about = "A Redis server")]
+#[derive(Parser, Debug)]
+#[clap(name = "mini-redis-server", version, author, about = "A Redis server")]
 struct Cli {
-    #[structopt(name = "port", long = "--port")]
+    #[clap(name = "port", long = "--port")]
     port: Option<String>,
```

Comment:
i know this was previously parsed as a string, but it occurs to me that while we're here, we _could_ change the port to 
```suggestion
    #[clap(name = "port", long = "--port")]
    port: Option<u16>,
```
and get clap to do the number parsing for us?